### DISCUSSION TITLE
[xState v5 beta] Typing actions in createMachine() does not work

### DISCUSSION THREAD

**Discussion by Wastl77** [statelyai/xstate#4153] [Q&A] [answered]
Hi,

i am trying to type my machine in xState v5 beta, it works for the events , but not for the actions.

This is my machine (shortened for readability):

```
export const notbetriebRootMachine = createMachine(
{
  types: {} as {
    events:
      | { type: 'FETCH-SUCCESS'; data: Prisma.ResourceCreateInput[] }
      | { type: 'FETCH-ERROR' }
      | { type: 'RESOURCE-EVENT'; callsign: string; eventType: string };
    actions: { type: 'logEvent'; params: { message: string } };
  },
  id: 'Notbetrieb Root',
    initial: 'initialState',
    states: {
      initialState: {
        invoke: {
          id: 'fetchMachine',
          src: fetchInitialDataMachine,
        },
        on: {
          'FETCH-SUCCESS': {
            target: 'ready',
            actions: [
              { type: 'logEvent', params: { message: 'fetch success' } },
              assign({
                ref: ({ event, spawn }) =>
                  event.data.map((res) => {
                    return spawn(resource, {
                      systemId: res.callsign,
                    });
                  }),
              }),
            ],
... Rest of the machine
    {
      actions: {
        logEvent: ({ params }) => console.log(params.message),
      },
  }
);
```

Note that the assign function is not yet implemented, but inlined.

In the action implementation it gives me the following error:

"logEvent is never used in the machine definition"

and the following Typescript-Definition on hover:

```
(property) logEvent: ({ params }: ActionArgs<MachineContext, {
    type: 'FETCH-SUCCESS';
    data: Prisma.ResourceCreateInput[];
} | {
    type: 'FETCH-ERROR';
} | {
    type: 'RESOURCE-EVENT';
    callsign: string;
    eventType: string;
}, ParameterizedObject>) => void
```

For the params argument it says:

"Property 'params' does not exist on type 'ActionArgs<MachineContext, { type: "FETCH-SUCCESS"; data: ResourceCreateInput[]; } | { type: "FETCH-ERROR"; } | { type: "RESOURCE-EVENT"; callsign: string; eventType: string; }, ParameterizedObject>'."

I think i have done everything exactly like in the docs for v5 beta described.

What am i missing here? Any help appreciated!

Thank you!

**Comment by davidkpiano** [comment, marked as answer]
Actions aren't strongly typed yet, but they will be soon.

**Reply by Wastl77** [reply]
OK, thank you for the quick reply!
